hamster_gtk/preferences/widgets/editable_file_chooser.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+
+# This file is part of 'hamster-gtk'.
+#
+# 'hamster-gtk' is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# 'hamster-gtk' is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with 'hamster-gtk'.  If not, see <http://www.gnu.org/licenses/>.
+
+"""This module provides file chooser widget."""
+
+# [FIXME]
+# Adding 'unicode_literals' raises encoding issues. This is a major sign we
+# have a unicode issue!
+from __future__ import absolute_import
+
+from gettext import gettext as _
+
+from gi.repository import Gtk
+from six import text_type
+
+from .config_widget import ConfigWidget
+
+
+class EditableFileChooser(Gtk.Grid, ConfigWidget):
+    """A file chooser that also has an entry for changing the path."""
+
+    # Required else you would need to specify the full module
+    # name in ui file
+    __gtype_name__ = 'EditableFileChooser'
+
+    def __init__(self):
+        """Initialize widget."""
+        super(Gtk.Grid, self).__init__()
+
+        self.entry = Gtk.Entry()
+        self.entry.set_hexpand(True)
+
+        self.button = Gtk.Button(_("Choose"))
+        self.button.connect('clicked', self.on_choose_clicked)
+
+        self.attach(self.entry, 0, 0, 1, 1)
+        self.attach(self.button, 1, 0, 1, 1)
+        self.connect('mnemonic-activate', self.on_mnemonic_activate)
+
+    def get_config_value(self):
+        """Return value of the entry."""
+        return self.entry.get_text()
```

Comment:
Should't this be wrapped in ``hamster.helpers._u()`` to make sure a unicode instance is returned regardless of python version?